MRML/vtkIGTLToMRMLPolyData.cxx

```diff
@@ -380,10 +380,9 @@ int vtkIGTLToMRMLPolyData::MRMLToIGTL(unsigned long event, vtkMRMLNode* mrmlNode
     
     //------------------------------------------------------------
     // Allocate Status Message Class
-    if (this->OutPolyDataMessage.IsNull())
-      {
-      this->OutPolyDataMessage = igtl::PolyDataMessage::New();
-      }
+    // Overwrite old message because we want to clear arrays from previous
+    // calls that will not be set anew.
+    this->OutPolyDataMessage = igtl::PolyDataMessage::New();
```

Comment:
Thanks for the comments, @lassoan and @christiana. 

The igtl::PolyDataMessage class has Clear() function to remove the previous data. Maybe we call this instead of calling New() every time? Still you can avoid some memory reallocations.